Source/DafnyCore/AST/IndentationFormatter.cs

```diff
@@ -0,0 +1,2393 @@
+  }
+
+  // For example, a closing brace
+  //
+  // var x: map<   T
+  //           ,
+  //               U
+  //               // beforeIndent
+  //           >   // this line indent's
+  //           //  after liene indent
+  // // not indented
+  private void SetClosingIndentedRegionAligned(IToken token, int beforeIndent, int indent) {
+    SetIndentations(token, beforeIndent, indent, indent);
+  }
+
+  /// For example, a closing brace
+  ///
+  /// if true {
+  ///   // indented
+  /// } // not indented
+  /// // not indented
+  private void SetClosingIndentedRegion(IToken token, int indent) {
+    SetIndentations(token, indent + SpaceTab, indent, indent);
+  }
+
+  /// For example, a semicolon
+  ///
+  /// var x := 2
+  ///   // indented
+  ///   ; // indented
+  /// // not indented
+  private void SetClosingIndentedRegionInside(IToken token, int indent) {
+    SetIndentations(token, indent + SpaceTab, indent + SpaceTab, indent);
+  }
+
+
+  /// For example, a "else" keyword for a statement
+  ///
+  /// if x == 2 {
+  /// }
+  /// // not indented
+  /// else // not indented
+  /// // not indented
+  /// {
+  /// }
+  /// // not indented
+  private void SetKeywordWithoutSurroundingIndentation(IToken token, int indent) {
+    SetIndentations(token, indent, indent, indent);
+  }
+
+  /// For example, an "equal" sign not followed by a space which indicates that things should be aligned
+  ///
+  /// var x, y :=    A()      //  The token ':=' is being considered
+  ///           ,    B();
+  /// ^indent and indentBefore
+  ///                ^rightIndent
+  ///           ^commaIndent
+  /// 
+  private void SetAlign(int indent, IToken token, out int rightIndent, out int commaIndent) {
+    SetIndentations(token, indent, indent);
+    rightIndent = GetRightAlignIndentAfter(token, indent);
+    commaIndent = GetNewTokenVisualIndent(token, indent) + token.val.Length - 1;
+    SetIndentations(token, after: rightIndent);
+  }
+
+  /// For example, a "then" keyword not followed by a newline, indicating that the content should be aligned with the first character.
+  ///
+  /// if X
+  /// then   proofY();
+  ///        result
+  /// ^indent
+  ///        ^ rightIndent
+  /// else Z
+  /// 
+  private void SetAlignOpen(IToken token, int indent) {
+    var rightIndent = GetRightAlignIndentAfter(token, indent);
+    SetIndentations(token, indent, indent, rightIndent);
+  }
+  #endregion
+
+  public string ReindentLeadingTrivia(IToken token, bool precededByNewline,
+    string indentationBefore, string lastIndentation) {
+    return Reindent(token, false, precededByNewline, indentationBefore, lastIndentation);
+  }
+
+  public string ReindentTrailingTrivia(IToken token, bool precededByNewline,
+    string indentationBefore, string lastIndentation) {
+    return Reindent(token, true, precededByNewline, indentationBefore, lastIndentation);
+  }
+
+  #region Override for implementing IIndentationFormatter
+  /// Given a space around a token (input),
```

Comment:
Could you fix the grammar in this comment and add examples of the trailingTrivia and !trailingTrivia case?

Reply:
I rephrased the entire comment to make it clearer, and added four examples.